### PR TITLE
Fix issue where genre wasn't showing and year is empty

### PR DESCRIPTION
**Pull request by fernandog**
Its not in my repo so other can help.
Seems the empty year is was causing the tag not showing

 I also can't understand why there's another minor empty tag

In this first two images I change html to show genres:

Working show:

Before:

After:
image

### CODE REVIEW

**Comment by fernandog** [comment]
@SickRage/collaborators

File: gui/slick/views/displayShow.mako [hunk 1]
```diff
@@ -126,8 +126,10 @@
                 <img src="${srRoot}/images/blank.png" class="country-flag flag-${country}" width="16" height="11" style="margin-left: 3px; vertical-align:middle;" />
         % endfor
     % endif
-    % if 'year' in show.imdb_info:
+    % if 'year' in show.imdb_info and len(show.imdb_info['year']) == 4:
                 <span>(${show.imdb_info['year']}) - ${show.imdb_info['runtimes']} minutes - </span>
+    % else:
+                <span>${show.imdb_info['runtimes']} minutes</span>
```
Comment (OmgImAlexis) [inline gui/slick/views/displayShow.mako]: Shouldn't this

````
% if 'year' in show.imdb_info and len(show.imdb_info['year']) == 4:
    <span>(${show.imdb_info['year']}) - ${show.imdb_info['runtimes']} minutes - </span>
% else:
    <span>${show.imdb_info['runtimes']} minutes</span>
% endif
````

be this?

````
<span>
% if 'year' in show.imdb_info and len(show.imdb_info['year']) == 4:
    (${show.imdb_info['year']}) - 
% endif
${show.imdb_info['runtimes']} minutes</span>
````

Comment (fernandog) [inline gui/slick/views/displayShow.mako]: @OmgImAlexis same result as image "After"

Comment (miigotu) [inline gui/slick/views/displayShow.mako]: `if 'year' in show.indb_info and show.imdb_info['year']:`

File: gui/slick/views/displayShow.mako [hunk 2]
```diff
@@ -139,7 +141,7 @@
 
             <div id="tags">
                 <ul class="tags">
-                    % if not show.imdbid and show.genre:
```
Comment (miigotu) [inline gui/slick/views/displayShow.mako]: `% if show.imdbid and show.genre:`

Comment (fernandog) [inline gui/slick/views/displayShow.mako]: I added that. but genre comes from trakt or imdb?

**Comment by fernandog** [comment]
I added @miigotu and @OmgImAlexis changes but still:

${show.genre[1:-1].split('|')} = [u'Action', u'Drama']

any clue wtf is going on?

image

**Comment by fernandog** [comment]
Seems IMDB genre that is empty. not Trakt.....

File: gui/slick/views/displayShow.mako [hunk 3]
```diff
@@ -139,7 +142,7 @@
...
+                    % if show.imdbid and show.genre:
                         % for genre in show.genre[1:-1].split('|'):
                             <a href="${anon_url('http://trakt.tv/shows/popular/?genres=', genre.lower())}" target="_blank" title="View other popular ${genre} shows on trakt.tv."><li>${genre}</li></a>
```
Comment (Thraxis) [inline gui/slick/views/displayShow.mako]: Can add a check here:

``` python
                         % for genre in show.genre[1:-1].split('|'):
                               % if genre.strip():
                             <a href="${anon_url('http://trakt.tv/shows/popular/?genres=', genre.lower())}" target="_blank" title="View other popular ${genre} shows on trakt.tv."><li>${genre}</li></a>                
```

**Comment by duramato** [comment]
And wab the issue @OmgImAlexis was talking of the imdb genre differing from trakt ?

**Comment by fernandog** [comment]
@miigotu @Thraxis @medariox 
we were checking YEAR to display GENRE.  makes no sense.

we can display genres from indexer OR imdb.
imdb genre has preference.

**Comment by fernandog** [comment]
@OmgImAlexis @duramato now imdb genres are used to create imdb genre link. no issues anymore

File: gui/slick/views/displayShow.mako [hunk 4]
```diff
@@ -139,14 +142,14 @@
...
             <div id="tags">
                 <ul class="tags">
-                    % if not show.imdbid and show.genre:
+                    % if not show.imdb_info['genres'] and show.genre:
                         % for genre in show.genre[1:-1].split('|'):
                             <a href="${anon_url('http://trakt.tv/shows/popular/?genres=', genre.lower())}" target="_blank" title="View other popular ${genre} shows on trakt.tv."><li>${genre}</li></a>
                         % endfor
                     % endif
-                    % if 'year' in show.imdb_info:
+                    % if show.imdb_info['genres']:
```
Comment (miigotu) [inline gui/slick/views/displayShow.mako]: I think the year check was just seeing if imdb info was loaded at all, but here you need to check if genres is a key in indb_info before accessing it:

`if 'genres' in show.imdb_info and show.imdb_info['genres']:`

**Comment by fernandog** [comment]
@miigotu done

**Pull request merged**